src/Sprite.js

```diff
@@ -481,6 +481,60 @@ export class Sprite extends SpriteBase {
     } while (t < 1);
   }
 
+  ifOnEdgeBounce() {
+    const nearestEdge = this.nearestEdge();
+    if (!nearestEdge) return;
+    const rad = this.scratchToRad(this.direction);
+    let dx = Math.cos(rad);
+    let dy = Math.sin(rad);
+    switch (nearestEdge) {
+      case Sprite.Edge.LEFT:
+        dx = Math.max(0.2, Math.abs(dx));
+        break;
+      case Sprite.Edge.RIGHT:
+        dx = -Math.max(0.2, Math.abs(dx));
+        break;
+      case Sprite.Edge.TOP:
+        dy = -Math.max(0.2, Math.abs(dy));
+        break;
+      case Sprite.Edge.BOTTOM:
+        dy = Math.max(0.2, Math.abs(dy));
+        break;
+    }
+    this.direction = this.radToScratch(Math.atan2(dy, dx));
+    const { x, y } = this.keepInFence(this.x, this.y);
+    this.goto(x, y);
+  }
+
+  keepInFence(newX, newY) {
+    // https://github.com/LLK/scratch-vm/blob/develop/src/sprites/rendered-target.js#L949
+    const fence = this.stage.fence;
+    const bounds = this._project.renderer.getTightBoundingBox(this);
+    bounds.left += newX - this.x;
+    bounds.right += newX - this.x;
+    bounds.top += newY - this.y;
+    bounds.bottom += newY - this.y;
```

Comment:
Are these `bounds.left += newX - this.x` etc. lines actually doing anything in this case? `newX === this.x` anyway so the deltas here are zero. I know this is directly pulled from Scratch 3.0, but it looks like effectively dead code, in which case no need to replicate it here!

OG implementation is in LLK/scratch-vm#257 (back from 2016!). Includes the same "+= delta" lines here, which even then were no-ops. tmickel specifically mentioned *"I also refactored the "keep on stage" logic into its own method, keepInFence, which can take an arbitrary bounding box to keep the clone inside. I figure this will be useful when we try to reproduce fence mode."* but they ended up using a separate implementation for fencing within the stage (uses [`getFencedPositionOfDrawable`](https://github.com/LLK/scratch-render/blob/791b2750cef140e714b002fd275b5f8434e6df9b/src/RenderWebGL.js#L1526) from scratch-render). This is just a hunch, but it may have been that the delta lines here would've been relevant if they'd used `keepInFence` for general fencing instead of just "if on edge, bounce", since these lines don't exist in [the Scratch 2.0 implementation](https://github.com/LLK/scratch-flash/blob/8bba2aef76d90162405909a343187bf3b9794f5b/src/primitives/MotionAndPenPrims.as#L363-L373).

Reply:
Agreed.  This feels like a good change!

Reply:
Implemented in 5186824